great_expectations/experimental/datasources/data_asset/data_connector/data_connector.py

```diff
@@ -0,0 +1,190 @@
+from __future__ import annotations
```

Comment:
This entire module was developed from scratch following the spirit of the relevant parts of the V3 interface design, but with the important difference that while in V3,  data connectors instantiate data assets and use the data asset configuration as overrides, in ZEP data assets are the primary object that instantiate data connectors and use them for efficient scanning of remote resources and for querying and sorting the corresponding data references encapsulated in `BatchDefinition` objects (a useful intermediate form).  The interface method signatures here for ZEP are simplified than those in V3 and qualitatively have different meaning.  Hence, new code.